mlflow/spark.py

```diff
@@ -226,6 +227,24 @@ def maybe_copy_from_local_file(cls, src, dst):
         _logger.info("Copied SparkML model to %s", dst)
         return dst
 
+    @classmethod
+    def maybe_copy_from_uri(cls, src_uri, dst_path):
+        """
+        Conditionally copy the file to the Hadoop DFS from the source uri.
+        In case the file is already on the Hadoop DFS do nothing.
+
+        :return: If copied, return new target location, otherwise return source uri.
+        """
+        import urllib
+        parsed_uri = urllib.parse.urlparse(src_uri)
+        try:
+            if parsed_uri.scheme and cls._fs().exists(cls._remote_path(src_uri)):
```

Comment:
When would we expect this to fail, and why would we expect that recursively calling the function again would succeed if this logic fails for the first set of arguments?

Reply:
1. I added the try/ catch is just in case.  I only tested this on Databricks with DBFS for DFS. It is possible that this would not work with some combination of  (uri  + spark version + hadoop fs implementation) and since this is only performance optimization we should not break any such use case. 
2. The function called outside of try / catch is different - it defaults to previous behavior of copying stuff locally and uploading it to a temporary path on DFS..

Reply:
Whoops, didn't catch the distinction. Thanks! Can we log the error if it fails?